ironic/tests/unit/drivers/modules/drac/test_management.py

```diff
@@ -423,17 +440,22 @@ def test_set_boot_device_called_with_unknown_boot_mode(
                           drac_mgmt.set_boot_device, self.node,
                           ironic.common.boot_devices.DISK, persistent=True)
 
+        mock_list_unfinished_jobs.assert_called_once_with(self.node)
         mock_validate_job_queue.assert_called_once_with(self.node)
         self.assertEqual(0, mock_client.change_boot_device_order.call_count)
         self.assertEqual(0, mock_client.set_bios_settings.call_count)
         self.assertEqual(0, mock_client.commit_pending_bios_changes.call_count)
 
     @mock.patch.object(drac_job, 'validate_job_queue', spec_set=True,
                        autospec=True)
+    @mock.patch.object(drac_job, 'list_unfinished_jobs', spec_set=True,
+                       autospec=True)
     def test_set_boot_device_with_invalid_job_queue(
-            self, mock_validate_job_queue, mock_get_drac_client):
+            self, mock_list_unfinished_jobs, mock_validate_job_queue,
+            mock_get_drac_client):
```

Comment:
I'm not exactly sure how this test case works...  In order for validate_job_queue to be called, list_unfinished_jobs must return a list containing at least one job that is not a BIOS job.  As it's written, I would think that the mocked list_unfinished_jobs would return None, and validate_job_queue would never be called.  Would you mind looking in to this?

Reply:
Hi Chris, above test case was for invalid job queue which raises error, I have added another positive test case to call validate_job_queue based on specified condition.

Reply:
I'm still not sure how this works.  It looks like it was meant to test an exception that occurs in validate_job_queue() in the original code.  In the new code though, list_unfinished_jobs() will be called, and it will return no jobs since it is mocked above.  It will then fall through to set_boot_device() and never call validate_job_queue().

Reply:
So with the 2 test cases below, and if you add one more that tests returning multiple jobs from list_unfinished_jobs, I'm thinking that this test case could be deleted.

Reply:
Okay, so I will delete existing test case - test_set_boot_device_with_invalid_job_queue . And will add another test case for multiple jobs from list_unfinished_jobs.

Reply:
Done